### PR TITLE
Allow users to create additional options

### LINKED ISSUE #111: Allow users to create additional options
#### Summary

We should allow users to create additional options in addition to the ones defined at the creation of the poll. There should be a Poll Setting for this. I didn't came up with a good name jet, maybe something like `--additional-options`. With this setting selected a new button "Add Option" would show on the left of "Delete Poll", after the options. This opens an interactive message dialogue where the user enters the option text.

#### Describe alternatives you've considered

1. We might allow this by default and add a setting to disable it.
2. Just allow the poll creator to add new options.

#### Additional context

See https://mattermost.atlassian.net/browse/MM-12790 for more infos about Interactive message dialogues.

### LINKED ISSUE #115: Prohibit polls with duplicate options
#### Summary
Users can create polls with duplicate options.

#### Steps to reproduce
1. Create a new poll via `/poll "Question" "A" "A" "A"`

#### Expected behavior
An error message gets shown.

#### Observed behavior (that appears unintentional)
A poll with three times the same options gets created.

### PR DESCRIPTION
**Pull request by hanzei**
Fixes #111 
Fixes #115

### CODE REVIEW

**Comment by hanzei** [comment]
@kaakaa Could you give this a first round of review? Tests are still missing but I'm quite fine with the code.

**Comment by kaakaa** [comment]
Thank you for sharing code. @hanzei 
Your code looks good 👍   Please go ahead.

File: server/plugin/command.go [hunk 1]
```diff
@@ -20,7 +20,8 @@ const (
 		"You can customise the options by typing `/%[1]s \"Question\" \"Answer 1\" \"Answer 2\" \"Answer 3\"`\n" +
 		"Poll Settings provider further customisation, e.g. `/%[1]s \"Question\" \"Answer 1\" \"Answer 2\" \"Answer 3\" --progress --anonymous`. The available Poll Settings are:\n" +
 		"- `--anonymous`: Don't show who voted for what at the end\n" +
-		"- `--progress`: During the poll, show how many votes each answer option got\n"
+		"- `--progress`: During the poll, show how many votes each answer option got\n" +
+		"- `--public-add-option`: Allow other users to add additonal options"
```
Comment (kaakaa) [inline server/plugin/command.go]: `other` is ambiguous, so `all` is better.

File: server/poll/poll.go [hunk 2]
```diff
@@ -37,22 +38,38 @@ func NewPoll(creator, question string, answerOptions, settings []string) (*Poll,
...
 	}
 	return &p, nil
 }
 
+// TODO: Add docs
+// AddAnswerOption
+func (p *Poll) AddAnswerOption(newAnswerOption string) error {
+	for _, answerOption := range p.AnswerOptions {
+		if answerOption.Answer == newAnswerOption {
+			return fmt.Errorf("dublicant options: %s", newAnswerOption)
```
Comment (kaakaa) [inline server/poll/poll.go]: `duplicated`?

File: server/plugin/api.go [hunk 3]
```diff
@@ -118,11 +127,144 @@ func (p *MatterpollPlugin) handleVote(w http.ResponseWriter, r *http.Request) {
...
+
+	post, appErr := p.API.GetPost(request.CallbackId)
+	if appErr != nil {
+		p.API.LogError("failed to get post", "err", err.Error())
+		p.SendEphemeralPost(request.ChannelId, request.UserId, commandGenericError)
+		w.WriteHeader(http.StatusOK)
+		return
+	}
+
+	answerOption, ok := request.Submission[addOptionKey].(string)
```
[the remark this answers is not in the document]
Comment (hanzei) [reply server/plugin/api.go]: Fix. This is now checked via `AddAnswerOption`.

**Comment by hanzei** [comment]
I also reverted the changes to `README.md`. We should not document unpublished changes there. Let's defer this until we release the next feature release.

**Pull request merged**